mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxCounterDAOTest.java

```diff
@@ -0,0 +1,152 @@
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.mail;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.mailbox.cassandra.CassandraId;
+import org.apache.james.mailbox.cassandra.modules.CassandraMailboxCounterModule;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.mail.model.impl.SimpleMailbox;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CassandraMailboxCounterDAOTest {
+    public static final int UID_VALIDITY = 15;
+    public static final CassandraId MAILBOX_ID = CassandraId.timeBased();
+
+    private CassandraCluster cassandra;
+    private CassandraMailboxCounterDAO testee;
+    private SimpleMailbox mailbox;
+
+    @Before
+    public void setUp() {
+        cassandra = CassandraCluster.create(new CassandraMailboxCounterModule());
+        cassandra.ensureAllTables();
+
+        testee = new CassandraMailboxCounterDAO(cassandra.getConf());
+
+        mailbox = new SimpleMailbox(new MailboxPath("#private", "user", "name"), UID_VALIDITY, MAILBOX_ID);
+    }
+
+    @After
+    public void tearDown() {
+        cassandra.clearAllTables();
+    }
+
+    @Test
+    public void countMessagesInMailboxShouldReturnEmptyByDefault() throws Exception {
+        assertThat(testee.countMessagesInMailbox(mailbox).join().isPresent()).isFalse();
+    }
+
+    @Test
+    public void countUnseenMessagesInMailboxShouldReturnEmptyByDefault() throws Exception {
+        assertThat(testee.countUnseenMessagesInMailbox(mailbox).join().isPresent()).isFalse();
+    }
+
+    @Test
+    public void incrementCountShouldAddOneWhenAbsent() throws Exception {
+        testee.incrementCount(MAILBOX_ID).join();
+
+        assertThat(testee.countMessagesInMailbox(mailbox).join().get()).isEqualTo(1);
```

Comment:
Forgot. Anyway, it was just to make exception more readable. It will have no effect on test. Just error instead of failure.